opendevin/runtime/browser/browser_env.py

```diff
@@ -47,14 +72,29 @@ def init_browser(self):
             raise BrowserInitException('Failed to start browser environment.')
 
     def browser_process(self):
-        env = gym.make(
-            'browsergym/openended',
-            task_kwargs={'start_url': 'about:blank'},
-            wait_for_user_message=False,
-            headless=True,
-            disable_env_checker=True,
-        )
+        if self.eval_mode:
+            logger.info('Creating browser env for evaluation purpose.')
+            import browsergym.webarena  # noqa F401 register webarena tasks as gym environments for eval
+            import browsergym.miniwob  # noqa F401 register miniwob tasks as gym environments
+
+            env = gym.make(self.browsergym_eval)
+        else:
+            env = gym.make(
+                'browsergym/openended',
+                task_kwargs={'start_url': 'about:blank', 'goal': 'PLACEHOLDER_GOAL'},
+                wait_for_user_message=False,
+                headless=True,
+                disable_env_checker=True,
+            )
         obs, info = env.reset()
+        # EVAL only: save the goal into file for evaluation
+        if self.eval_mode:
```

Comment:
Prefer to find a better way to record the intermedia data in the future, to avoid evaluation and runtime logic get mixed. 

But it is fine for me to do this quick implementation here.

Reply:
I agree. This is ugly for now - I used filesystem to do IPC...

A much better solution in my opinion to better support BrowserGym benchmarks are allowing BrowserEnv (running in a separate process) to access the `EventStream` that our main process have. If we can send the initial `MessageAction` then these goal sharing code and reward saving code can be just removed - they can be communicated to the main process and we can handle all these eval-specific things in evaluation codebase only.

Maybe somehow in the future if we have a unified MQ this might be easier? Now maybe we have to expose something to allow for better IPC between browser environment to the main process where we handle all the events. Love your thoughts, but I think this might be worth a new issue?

Reply:
>  I think this might be worth a new issue?

Yes, I agree. Better to do it in future PR.